### PR TITLE
Contact form (both non-admin and admin)

### PR DESCRIPTION
**Pull request by sanjayss34**

### CODE REVIEW

File: app/main/views.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import json
 
-from flask import render_template, request
+from flask import render_template, url_for, request
 from flask.ext.login import login_required
 
 from .. import db
-from ..models import EditableHTML, Resource
```
Comment (anniemeng) [inline app/main/views.py]: Resource is still used on line 23

File: app/contact/views.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        category.name = form.name.data
+        db.session.add(category)
+        try:
+            db.session.commit()
+            flash('Name for category {} successfully changed to {},'
+                .format(old_name, category.name),
+                'form-success')
+        except IntegrityError:
+            db.session.rollback()
+            flash('Database error occurred. Please try again', 'form-error')
```
Comment (anniemeng) [inline app/contact/views.py]: nit: period at end of sentence

File: app/contact/views.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    category = ContactCategory.query.get(category_id)
+    if category is None:
+        abort(404)
+    old_name = category.name
+    form = EditCategoryNameForm()
+    if form.validate_on_submit():
+        if ContactCategory.query.filter(ContactCategory.name == form.name.data).first() is not None:
+            flash('Category {} already exists.'.format(form.name.data), 'form-error')
+            return render_template(
+                                'contact/manage_category.html',
```
Comment (anniemeng) [inline app/contact/views.py]: nit: put name of template on same line as `render_template` and then align variables with that

same for other `render_template` functions in file

File: app/contact/views.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            recipient=contact_email,
+            subject=form.category.data,
+            template='contact/email/contact',
+            name=form.name.data,
+            email=form.email.data,
+            message=form.message.data
+        )
+        return redirect(url_for('main.index'))
+    category_form = ContactCategoryForm()
+    if category_form.validate_on_submit():
```
Comment (sanjayss34) [inline app/contact/views.py]: Added in the check, but for some reason the flash isn't showing up

File: app/templates/contact/index.html [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            'info': get_flashed_messages(category_filter=['form-info']),
+            'success': get_flashed_messages(category_filter=['form-success'])
+            }
+        %}
+        {% if current_user.is_admin() %}
+            <h3>Edit Contact Description</h3>
+        {% endif %}
+        {{ page.render_inline_editor(editable_html_obj, current_user) }}
+        {% if not current_user.is_admin() %}
+
```
Comment (sanjayss34) [inline app/templates/contact/index.html]: Tried doing this with ui horizontal divider but didn't work without adding text. How can I do this?

**Comment by sanjayss34** [comment]
All changes other than those noted above have been made

File: app/templates/contact/index.html (hunk 5 above)
Comment (anniemeng) [inline app/templates/contact/index.html]: what if you just put ui divider?

File: app/contact/views.py (hunk 4 above)
Comment (anniemeng) [inline app/contact/views.py]: try returning render template here? or are there any errors you get?

File: app/contact/views.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            flash('Category {} already exists.'.format(form.name.data), 'form-error')
+            return render_template(
+                                'contact/manage_category.html',
+                                category=category,
+                                form=form)
+        category.name = form.name.data
+        db.session.add(category)
+        try:
+            db.session.commit()
+            flash('Name for category {} successfully changed to {},'
```
Comment (anniemeng) [inline app/contact/views.py]: bump: add quotes for both `{}`

**Comment by sanjayss34** [comment]
Added the quotes and changed comma to period

**Comment by anniemeng** [comment]
does the flash show up now? also let's merge in master and resolve conflicts

**Comment by sanjayss34** [comment]
No, I never got the flash to work.

**Comment by anniemeng** [comment]
@sanjayss34 next time before we merge we want to make sure that the circleci tests pass

**Pull request merged**